0_global_functions.R

```diff
@@ -315,4 +315,15 @@ check_valid_cfg <- function(cfg){
   stopifnot(cfg$project_internal$twodii_internal %>% is.logical() == T)
   
   invisible(cfg)
+}
+
+
+#write error log for input portfolio - msg should be a string containing the error message
+write_log <- function(msg, ...) {
```

Comment:
Question perhaps whether these should be labelled "error_messages" or more a Warning message. But this isn't too critical.